### DISCUSSION TITLE
Data streaming functionality in near real time (above one second)

### DISCUSSION THREAD

**Discussion by jfernandezsqs** [International-Data-Spaces-Association/DataspaceConnector#112] [Q&A] [answered]
I have deployed two Dataspace Connector version 6.2.0 (Connector A acting as provider and Connector B acting as consumer).
I have followed the documentation and obtained the data file offered by Connector A. (size of the file around 50MB).
For the communication between connectors and the resource obtained it is expected to get the data in near real time (under one second).
Currently, the response time is over one second for a file of that size.

![image](https://user-images.githubusercontent.com/82888094/148534524-3a8823bd-aff0-47d6-a864-ae510ad42a25.png)

In the communication between connectors, is this an adequate response time for obtaining this file of data?

**Comment by juliapampus** [comment, marked as answer]
There is no general answer to this question. The speed of data transfer depends on many factors: i.a. the reaction time of the connected data source, the network velocity, the execution environment where the DSC is running (e.g. processor, RAM). Depending on the setup, the data transfer can thus be influenced and also optimized.
